### DISCUSSION TITLE
Dynamic animation duration when working with map

### DISCUSSION THREAD

**Discussion by meysamzi** [tailwindlabs/tailwindcss#8889] [Help]
Hi there,
I'm trying to change the duration of my animation when working with map 
Here is my code

    <div className="grid gap-3 md:grid-cols-2 lg:grid-cols-3" ref={ref}>
      {lists.map((item,**index**) => (
        <div key={item.id} className={`cursor-pointer group ${inView && `**animate-[rotate_${index}s]`}**`}>
          <div className="relative w-full h-[250px] lg:h-[350px]">
            <Image src={`/images/${item.imgUrl}`}
              className="rounded-md object-cover"
              layout={'fill'}
            />
            <div className="absolute inset-0 group-hover:h-full bg-[#a9b50160] transition-[height] duration-[0.4s] ease-in-out overflow- 
             hidden h-0 flex flex-col gap-y-8 items-center justify-center text-white text-lg">
              <FontAwesomeIcon icon={faEye} />
              <Link href={item.href}>
                <a className="hover:font-bold">بیشتر</a>
              </Link>
            </div>
          </div>
        </div>
      ))}
    </div>

**Comment by arisac** [comment]
That's not gonna work with fully dynamic value on the frontend. TailwindCSS is CSS Classes in typical build.

For fully dynamic values, you can always use inline style, `style={{}}` for react.

But if you want to use animation, it can be complicated with pure css styles to write with. Since your project is react, you can just use `framer-motion` lib to do that, it's quite simple for your case:

install framer-motion: `npm install framer-motion` or `yarn add framer-motion`

use framer-motion:

```jsx
import { motion } from 'framer-motion'
...
<motion.div
  initial={{ rotate: 0 }}
  animate={{ rotate: value }}
  transition={{ duration: 0.5 }}
  className="..."
>
  ...
</motion.div>
```

more about framer motion: https://www.framer.com/docs/introduction/
